### PR TITLE
HWKALERTS-89 Normalize Incoming Data Format

### PR DESCRIPTION
**Pull request by jshaughn**
- remove Availability, NumericData StringData and MixedData
- redefine Data to be used as the sole data format
  - let the ConditionEval impls deal with any conversion
  - provide convenience factory methods for when caller knows
    the data type he wants to provide.
  - Note, support for multi-value conditions (i.e. the 'values'
    field) is anticipated but not yet a reality.
- Make AvailabilityType a 1st class citizen
- Note: REST API CHANGE in sendData endpoint, but this is
  not [yet] used in Hawkular code outside of alerts.
- Update bus integration code (topic listeners) to use new
  mechanism

### CODE REVIEW

**Comment by jshaughn** [comment]
Lucas, review when you get the chance.  Remember, let's not merge anything until we release 0.4.1 to support M5.

File: hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/data/Data.java [hunk 1]
```diff
@@ -45,45 +38,95 @@
...
-    protected Object value;
+    /** For single-value condition types. Null otherwise */
+    @JsonInclude(Include.NON_EMPTY)
+    protected String value;
 
-    @JsonInclude
-    protected Type type;
+    /** For multi-value condition types. Null otherwise. See the condition type for expected key-value information */
+    @JsonInclude(Include.NON_EMPTY)
+    protected Map<String, String> values;
```
Comment (lucasponce) [inline hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/data/Data.java]: Minor comment. I think is just a matter of style, for me having Data.value and Data.values for single or multiple values it less clear that having just a values and perhaps a helper method for single values.

**Comment by lucasponce** [comment]
Hi Jay,
I've reviewed the PR and after the chat about the context I don't see any major comment.
I've put a small comment at your convenience.

One thing, as this PR has some new things related to API (sendData) I'm thinking to merge in a new branch "called Events" for future tasks.

We can maintain master for 0.4.1 or 0.4.x in case some last minute error requires small fix for Hawkular MS5.

**Comment by jshaughn** [comment]
Closing, re-opened against Events branch

**Comment by jshaughn** [comment]
V

**Pull request closed without merging**